pkg/api/v1beta2/dynakube/dynakube_conversion.go

```diff
@@ -0,0 +1,160 @@
+
+	// Status
+	dst.Status.Conditions = src.Status.Conditions
+
+	if dst.Annotations == nil {
+		dst.Annotations = map[string]string{}
+	}
+
+	if !src.Spec.MetaDataEnrichment.Enabled {
+		dst.Annotations[dynakube.AnnotationFeatureMetadataEnrichment] = "false"
+	}
+
+	dst.Annotations[dynakube.AnnotationFeatureApiRequestThreshold] = strconv.FormatInt(int64(src.Spec.DynatraceApiRequestThreshold), 10)
+
+	if hostMonitoring := src.Spec.OneAgent.HostMonitoring; hostMonitoring != nil {
+		if hostMonitoring.SecCompProfile != "" {
+			dst.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile] = hostMonitoring.SecCompProfile
+		}
+	}
+
+	if src.Spec.OneAgent.CloudNativeFullStack != nil {
+		if !isLabelSelectorEmpty(src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector) {
+			matchExpressions := src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector.MatchExpressions
+			matchLabels := src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector.MatchLabels
+
+			dst.Spec.NamespaceSelector = v1.LabelSelector{
+				MatchExpressions: matchExpressions,
+				MatchLabels:      matchLabels,
+			}
+		}
+	} else if src.Spec.OneAgent.ApplicationMonitoring != nil {
+		if !isLabelSelectorEmpty(src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector) {
+			matchExpressions := src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector.MatchExpressions
+			matchLabels := src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector.MatchLabels
+
+			dst.Spec.NamespaceSelector = v1.LabelSelector{
+				MatchExpressions: matchExpressions,
+				MatchLabels:      matchLabels,
+			}
+		}
+	}
+
+	dst.Status.OneAgent.Instances = map[string]dynakube.OneAgentInstance{}
+
+	for key, value := range src.Status.OneAgent.Instances {
+		tmp := dynakube.OneAgentInstance{
+			PodName:   value.PodName,
+			IPAddress: value.IPAddress,
+		}
+		dst.Status.OneAgent.Instances[key] = tmp
+	}
+
+	dst.Status.OneAgent.Version = src.Status.OneAgent.Version
+
+	dst.Status.Phase = src.Status.Phase
+	dst.Status.UpdatedTimestamp = src.Status.UpdatedTimestamp
+
+	return nil
+}
+
+func isLabelSelectorEmpty(selector v1.LabelSelector) bool {
+	return len(selector.MatchExpressions) == 0 && len(selector.MatchLabels) == 0
+}
+
+// ConvertFrom converts v1beta1 to v1beta2.
+func (dst *DynaKube) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*dynakube.DynaKube)
+	dst.ObjectMeta = src.ObjectMeta
+
+	// DynakubeSpec
+	dst.Spec.APIURL = src.Spec.APIURL
+	dst.Spec.Tokens = src.Spec.Tokens
+	dst.Spec.CustomPullSecret = src.Spec.CustomPullSecret
+	dst.Spec.SkipCertCheck = src.Spec.SkipCertCheck
+	dst.Spec.Proxy = (*DynaKubeProxy)(src.Spec.Proxy)
+	dst.Spec.TrustedCAs = src.Spec.TrustedCAs
+	dst.Spec.NetworkZone = src.Spec.NetworkZone
+	dst.Spec.EnableIstio = src.Spec.EnableIstio
+
+	if src.Annotations == nil {
+		src.Annotations = map[string]string{}
+	}
+
+	if !dst.Spec.MetaDataEnrichment.Enabled {
+		src.Annotations[dynakube.AnnotationFeatureMetadataEnrichment] = "false"
+	}
```

Comment:
I think you did a mix up here, the `v1beta2 => v1beta1` looks good, but here you are creating the `dst` so checking it makes little sense

Reply:
The whole `v1beta1 => v1beta2` looks a bit like copy-pasta from the `v1beta2 => v1beta1`, and while certain actions are the same, the more complicated ones are not, here you should be removing the annotations and not removing them

Reply:
should be done https://github.com/Dynatrace/dynatrace-operator/pull/3092/commits/4dba19d3d27b6550647c1382a1ddd444947aae64